### PR TITLE
Python 3: fix TLS layer

### PR DESCRIPTION
**Pull request by gpotter2**
This PR fixes netaccess tests for python 3

### CODE REVIEW

**Comment by gpotter2** [comment]
This PR should now be working

File: scapy/layers/tls/cert.py [hunk 1]
```diff
@@ -279,9 +279,9 @@ def fill_and_store(self, modulus=None, modulusLen=None, pubExp=None):
     def import_from_tuple(self, tup):
         # this is rarely used
         e, m, mLen = tup
-        if isinstance(m, str):
+        if isinstance(m, (str, bytes)):
```
Comment (p-l-) [inline scapy/layers/tls/cert.py]: I think `isinstance(m, bytes)` should be enough.

File: scapy/layers/tls/cert.py [hunk 2]
```diff
@@ -279,9 +279,9 @@ def fill_and_store(self, modulus=None, modulusLen=None, pubExp=None):
...
             m = pkcs_os2ip(m)
-        if isinstance(e, str):
+        if isinstance(e, (str, bytes)):
```
Comment (p-l-) [inline scapy/layers/tls/cert.py]: Same here.

File: scapy/layers/tls/cert.py (hunk 1 above)
Comment (gpotter2) [reply scapy/layers/tls/cert.py]: Updated

**Comment by gpotter2** [comment]
@p-l- This patch isn't working on python 3.3 (works on python 3.3+). Should I investigate or mark it as unavailable.
Otherwise any ideas ?

File: scapy/layers/tls/crypto/cipher_block.py [hunk 3]
```diff
@@ -44,14 +44,14 @@ class _BlockCipher(six.with_metaclass(_BlockCipherMetaclass, object)):
 
     def __init__(self, key=None, iv=None):
         self.ready = {"key": True, "iv": True}
-        if key is None:
+        if not key:
```
Comment (p-l-) [inline scapy/layers/tls/crypto/cipher_block.py]: Is this needed? Why? Replacing `if X is None` by `if not X` or `if X is not None` by `if X` should be done only if necessary.

Same questions for other similar replacements.

Comment (gpotter2) [reply scapy/layers/tls/crypto/cipher_block.py]: You’re right. I used regex to replace all of those at once. Will rewind those changes tonight

**Comment by gpotter2** [comment]
@p-l- Updated

File: scapy/layers/tls/keyexchange.py [hunk 4]
```diff
@@ -570,7 +570,7 @@ def fill_missing(self):
             pubkey = s.server_kx_privkey.public_key()
             self.point = pubkey.public_numbers().encode_point()
         # else, we assume that the user wrote the server_kx_privkey by himself
-        if self.pointlen is None:
+        if self.pointlen  is None:
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply scapy/layers/tls/keyexchange.py]: Well i must be tired

**Pull request merged**